applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClient.java

```diff
@@ -60,242 +60,273 @@ public ConfigurationClient(String connectionString, HttpPipeline pipeline) {
+        // Closest to API goes first, closest to wire goes last.
+        final List<HttpPipelinePolicy> policies = new ArrayList<>();
+
+        policies.add(new UserAgentPolicy(String.format("Azure-SDK-For-Java/%s (%s)", SDK_NAME, SDK_VERSION)));
+        policies.add(new RequestIdPolicy());
+        policies.add(new RetryPolicy());
+        policies.add(new ConfigurationCredentialsPolicy(credentials));
+
+        return policies;
     }
 
     /**
-     * Sets key value. Label value for the configurationSetting is optional, if not specified or label=%00 it implies null label.
-     * @param configurationSetting key and value to set
+     * Adds a configuration value in the service if that key and label does not exist.
+     * <p>
+     * The label value for the ConfigurationSetting is optional. If not specified, the
+     * {@link ConfigurationSetting#NULL_LABEL} is used.
+     *
+     * @param configurationSetting The key, value, and label to set.
      * @return ConfigurationSetting that was created or updated
+     * @throws com.microsoft.azure.v3.CloudException when a ConfigurationSetting with the same key and label exists.
      */
-    public Mono<RestResponse<ConfigurationSetting>> setKeyValue(ConfigurationSetting configurationSetting, ETagFilter filter) {
+    public Mono<RestResponse<ConfigurationSetting>> addKeyValue(ConfigurationSetting configurationSetting) {
         Validator.validate(configurationSetting);
-        KeyValueCreateUpdateParameters parameters = new KeyValueCreateUpdateParameters().withValue(configurationSetting.value())
+        KeyValueCreateUpdateParameters parameters = new KeyValueCreateUpdateParameters()
+                .withValue(configurationSetting.value())
                 .withContentType(configurationSetting.contentType())
                 .withTags(configurationSetting.tags());
 
-        if (filter != null) {
-            return service.setKey(baseUri.toString(), configurationSetting.key(), configurationSetting.label(), parameters, filter.ifMatch(), filter.ifNoneMatch());
-        }
-        return service.setKey(baseUri.toString(), configurationSetting.key(), configurationSetting.label(), parameters, null, null);
+        return service.setKey(baseUri.toString(), configurationSetting.key(), configurationSetting.label(), parameters, null, getETagValue(ETAG_ANY));
     }
 
     /**
-     * Gets the ConfigurationSetting object for the specified key and KeyValueFilter2 parameters.
-     * @param key the key being retrieved
-     * @param filter options for the request
-     * @return ConfigurationSetting object
+     * Creates or updates a configuration value in the service.
+     * <p>
+     * If {@link ConfigurationSetting#etag()} is specified, the configuration value is updated if the current setting's
+     * etag matches. If the etag's value is equal to {@link ConfigurationClient#ETAG_ANY}, the setting will always be
+     * updated.
+     *
```

Comment:
Just from a javadoc style point of view, I'd like to see an empty line before the <p> element, and I'd like to also see a closing </p> element at the end. JavaDoc gets stricter in later releases about correctness of HTML, so we should start now.